client/client_state.cpp

```diff
@@ -413,6 +411,31 @@ static void set_client_priority() {
 #endif
 }
 
+// return the compile target of the BOINC client
+// (not necessarily the same has the host architecture)
+// TODO: flesh this out for Linux (architecture, bitness)
+//
+static const char* client_target() {
+#ifdef _WIN32
+#ifdef _WIN64
+    return " for Windows 64-bit";
+#endif
+    return " for Windows 32-bit";
+#endif
+
+#ifdef __APPLE__
+    return " for Mac OS X";
+#endif
+
+#if LINUX_LIKE_SYSTEM
+    return " for Linux";
+#endif
+
+#ifdef ANDROID
+    return " for Android";
+#endif
+}
+
```

Comment:
Do we need a default return type of " for Unknown" or some message in the event that none of the flags _WIN32/_WIN64/__APPLE__/LINUX_LIKE_SYSTEM/ANDROID is set?  I don't know all the cases that things are built and if not having one of these set causes problems elsewhere in the code, but it seems like there should be a default to handle that case.  

Note:  If the code absolutely requires one of these flags to be set, then I withdraw my comment.  However, if the intention is to be able to build the client without one of these flags then a default should exist.

Reply:
Additional comment, based on Juha's comments above, the use of HOSTTYPE in the default case would be better than "Unknown"